src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -690,9 +690,15 @@ void MutationsInterpreter::prepare(bool dry_run)
                 {
                     if (column.default_desc.kind == ColumnDefaultKind::Materialized)
                     {
+                        auto type_literal = std::make_shared<ASTLiteral>(column.type->getName());
+
+                        auto materialized_column = makeASTFunction("_CAST",
```

Comment:
Will this work properly for a scenario where c has Bool type and we do update b=0.1?
Because if we set b=0.1 it will store 0 in b and propagate to c as `false` and but if we cast 0.1 to Bool it is `true`
```
davenger-xps :) select CAST(0.1 as Int64)

SELECT CAST(0.1, 'Int64')

Query id: dd359c98-304f-4493-9702-d18178b69d6f

┌─CAST(0.1, 'Int64')─┐
│                  0 │
└────────────────────┘

1 row in set. Elapsed: 0.001 sec. 

davenger-xps :) select CAST(0.1 as Bool)

SELECT CAST(0.1, 'Bool')

Query id: 870fd862-2336-44bb-a50d-156486e7d7f3

┌─CAST(0.1, 'Bool')─┐
│ true              │
└───────────────────┘

1 row in set. Elapsed: 0.001 sec. 

davenger-xps :) select CAST(CAST(0.1 as Int64) as Bool)

SELECT CAST(CAST(0.1, 'Int64'), 'Bool')

Query id: a5291492-4ee4-4384-a308-042a86e4b98f

┌─CAST(CAST(0.1, 'Int64'), 'Bool')─┐
│ false                            │
└──────────────────────────────────┘

```

Reply:
I am not sure what scenario you see as a proper one, but currently, it works as `0.1 -> 0 -> false` and looks like the desired one to me